asyncscope.md

```diff
@@ -607,6 +608,82 @@ class Camera {
 }
 ```
 
+## Recursively spawning work until completion
+Below are two ways you could recursively spawn work on a scope using `let_with_async_scope` or `counting_scope`. Both
+examples are perfectly valid ways to ensure that all new work spawned from previous work will complete.
+
+:::cmptable
+
+### let_with_async_scope
+```cpp
+struct tree {
+  std::unique_ptr<tree> left;
+  std::unique_ptr<tree> right;
+  int data;
+};
+
+auto process(ex::scheduler sch, ex::counting_scope& scope, tree& t) {
+  return ex::schedule(sch) | then([sch, &]() {
+    if (t.left)
+      ex::spawn(scope, process(sch, scope, t.left.get()));
+    if (t.right)
+      ex::spawn(scope, process(sch, scope, t.right.get()));
+    do_stuff(t.data);
+  });
```

Comment:
In order to be able to `spawn()` the result of `process(…)`, it has to return a sender that never fails, which means, at a minimum, that `then`'s lambda has to be `noexcept`. In principle, the `schedule(sch)` sender could fail, too, though; maybe we need to wrap the whole pipeline in a `let_error` that swallows exceptions?
```suggestion
  return ex::schedule(sch) | then([sch, &]() noexcept {
    if (t.left)
      ex::spawn(scope, process(sch, scope, t.left.get()));
    if (t.right)
      ex::spawn(scope, process(sch, scope, t.right.get()));
    do_stuff(t.data);
  });
```